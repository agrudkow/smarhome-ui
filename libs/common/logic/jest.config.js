module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
