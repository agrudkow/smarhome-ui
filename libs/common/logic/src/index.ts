export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
