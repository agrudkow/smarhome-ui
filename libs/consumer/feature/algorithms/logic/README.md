# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
