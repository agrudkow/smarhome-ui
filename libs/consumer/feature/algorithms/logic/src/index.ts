export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
