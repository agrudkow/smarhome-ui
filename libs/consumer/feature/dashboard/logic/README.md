# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
