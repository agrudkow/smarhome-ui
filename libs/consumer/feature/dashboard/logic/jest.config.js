module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
