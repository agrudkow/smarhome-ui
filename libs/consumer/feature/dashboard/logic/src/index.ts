export * from './lib/consumer-feature-dashboard-logic';
