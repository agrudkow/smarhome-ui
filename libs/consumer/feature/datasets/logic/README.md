# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
