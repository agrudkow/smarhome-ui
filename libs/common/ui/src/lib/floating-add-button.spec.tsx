import React from 'react';
import { customRender } from '@smarthome/common/logic';
import FloatingAddButton from './floating-add-button';

describe(' FloatingAddButton', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(
      <FloatingAddButton onClick={jest.fn()} hoverText="" />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = customRender(
      <FloatingAddButton onClick={jest.fn()} hoverText="" />
    );
    expect(baseElement.firstChild).toMatchSnapshot();
  });

  it('should set hover text', () => {
    const hoverText = 'Test hover text';
    const { getByText } = customRender(
      <FloatingAddButton onClick={jest.fn()} hoverText={hoverText} />
    );
    expect(getByText(hoverText)).toBeTruthy();
  });
});
