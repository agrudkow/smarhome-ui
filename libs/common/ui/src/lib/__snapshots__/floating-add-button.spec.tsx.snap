// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` FloatingAddButton should match snapshot 1`] = `
.c0 {
  position: fixed;
  right: 20px;
  bottom: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  width: 56px;
  height: 56px;
  border-radius: 26px;
  overflow: hidden;
  opacity: 0.6;
  -webkit-transition: color 0.3s ease-in-out,background-color 0.3s ease-in-out, opacity 0.3s ease-in-out,width 0.3s ease-in-out;
  transition: color 0.3s ease-in-out,background-color 0.3s ease-in-out, opacity 0.3s ease-in-out,width 0.3s ease-in-out;
  background-color: #1BBDCE;
  color: #102128;
}

.c2 .add-icon {
  -webkit-transition: width 0.3s ease-in-out;
  transition: width 0.3s ease-in-out;
}

.c2:hover {
  background-color: #1BBDCE;
  color: #102128;
  width: 140px;
  opacity: 1;
}

.c2:hover .add-icon {
  width: 0;
}

.c3 {
  overflow: hidden;
  width: 0;
  white-space: nowrap;
  font-weight: 550;
  -webkit-transition: width 0.3s ease-in-out;
  transition: width 0.3s ease-in-out;
}

.c1:hover .c3 {
  width: 140px;
}

<div>
  <div
    class="c0"
  >
    <button
      aria-label="add"
      class="MuiButtonBase-root MuiFab-root c1 c2 MuiFab-primary"
      tabindex="0"
      textwidth="140"
      type="button"
    >
      <span
        class="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root add-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        <span
          class="c3"
        />
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
